x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/index/IndexAuditTrail.java

```diff
@@ -381,7 +390,7 @@ void updateCurrentIndexMappingsIfNecessary(ClusterState state) {
             IndexMetaData indexMetaData = indices.get(0);
             MappingMetaData docMapping = indexMetaData.mapping("doc");
             if (docMapping == null) {
-                if (indexToRemoteCluster || state.nodes().isLocalNodeElectedMaster()) {
+                if (indexToRemoteCluster || state.nodes().isLocalNodeElectedMaster() || hasStaleMessage()) {
                     putAuditIndexMappingsAndStart(index);
```

Comment:
bug 2.
master will only update the next rolled over index, if there are no events (on master) for the old index. However, there could be events of new nodes for the old index, whose template will never be updated.

Reply:
I'd rather have the master issue a mapping update for the previous index or indices. What do you think?

Reply:
I've been entertaining this idea too. I've chosen this alternative because the code is much simpler. In the obverse, the previous index might not exist or the rollover setting might have changed during the restart, which I think is too complex to code in an already complex mayhem. 
I think the surface for the dreaded cluster update storm is acceptable, as there is a very good chance the non-master will not be in sync during the repeated start retrials.